packages/next-auth/src/providers/notion.js

```diff
@@ -0,0 +1,93 @@
+export default function Notion(options) {
+    const {
+        clientId = "",
+        clientSecret = "",
+        notionVersion = "2022-02-22"
+    } = options
+
+    return {
```

Comment:
We're trying to remove plain JS from our codebase and move everything to TS (both for security and improved DX). 

I re-wrote this file in TS:
```ts
import { TokenSetParameters } from "openid-client"
import type { OAuthConfig, OAuthUserConfig } from "."

interface NotionTokenSet extends TokenSetParameters {
  owner?: {
    user: {
      id: string
    }
  }
}

interface NotionTokenSetParams {
  tokens: NotionTokenSet
}

export interface NotionProfile extends Record<string, any> {
  id: string
  name: string
  avatar_url: string
  person: {
    email: string
  }
}

export interface NotionProviderOptions {
  notionVersion: string
}

export default function Notion<P extends NotionProfile>(
  options: OAuthUserConfig<P> & NotionProviderOptions
): OAuthConfig<P> {
  return {
    id: "notion",
    name: "Notion",
    type: "oauth",
    authorization: {
      url: "https://api.notion.com/v1/oauth/authorize",
      params: {
        owner: "user",
      },
    },
    token: "https://api.notion.com/v1/oauth/token",
    userinfo: {
      request: async ({ tokens }: NotionTokenSetParams) => {
        const { access_token = "" } = tokens

        if (!access_token) {
          throw new Error("Notion Provider: No access token received")
        }

        const res = await fetch(
          "https://api.notion.com/v1/users/" + tokens.owner?.user.id,
          {
            headers: {
              Accept: "application/json",
              "Content-Type": "application/json",
              "Notion-Version": options.notionVersion || "2022-02-22",
              Authorization: "Bearer " + access_token,
            },
          }
        )

        if (!res.ok) {
          throw new Error("Something went wrong while trying to get the user")
        }

        const result = await res.json()
        return result
      },
    },
    profile: (profile) => ({
      id: profile.id,
      name: profile.name,
      email: profile.person.email,
      image: profile.avatar_url,
    }),
    ...options,
  }
}
```
Can you commit this instead?

I see **Notion** doesn't seem to have a **Well Known** endpoint 🙄 ( cc. @balazsorban44 ), but the options can be much simplified as per my example.

Note that internally we use `OpenID-client` to handle the OAuth callback for us, and this already uses **Basic Auth** for the token endpoint. However, I see we call it supplying a user ID `tokens.owner?.user.id` 🤔 

Could you try simplifying it even further with just:
```ts
import { TokenSetParameters } from "openid-client"
import type { OAuthConfig, OAuthUserConfig } from "."

export interface NotionProfile extends Record<string, any> {
  id: string
  name: string
  avatar_url: string
  person: {
    email: string
  }
}

export interface NotionProviderOptions {
  notionVersion: string
}

export default function Notion<P extends NotionProfile>(
  options: OAuthUserConfig<P> & NotionProviderOptions
): OAuthConfig<P> {
  return {
    id: "notion",
    name: "Notion",
    type: "oauth",
    authorization: {
      url: "https://api.notion.com/v1/oauth/authorize",
      params: {
        owner: "user",
      },
    },
    token: "https://api.notion.com/v1/oauth/token",
    userinfo: "https://api.notion.com/v1/users",
    profile: (profile) => ({
      id: profile.id,
      name: profile.name,
      email: profile.person.email,
      image: profile.avatar_url,
    }),
    ...options,
  }
}
```
and see if it works? 🤞🏽